postgres.go

```diff
@@ -75,6 +165,73 @@ func (db DBConf) dump(sb s3Backend, keyPath string) error {
 	return nil
 }
 
+// Basebackuprestore function:
+// - gets the key to decrypt the pg_data
+// - decrypts and decompress the data
+// - untar the data
+// - puts the db copy in the running container
+func (db DBConf) baseBackupRestore(sb s3Backend, keyPath, backupTar string) error {
+	localTar, err := os.Create("/home/backup.tar")
+	if err != nil {
+		log.Errorf("Error in creating file: %v", err)
+	}
+
+	fr, err := sb.NewFileReader(backupTar)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	defer fr.Close()
+
+	key := getKey(keyPath)
+	r, err := newDecryptor(key, fr)
+	if err != nil {
+		log.Error("Could not initialise decryptor", err)
+		return err
+	}
+
+	d, err := newDecompressor(r)
+	if err != nil {
+		log.Errorf("Could not initialise decompressor: %v", err)
+		return err
+
+	}
+
+	_, err = io.Copy(localTar, d)
+	if err != nil {
+		log.Errorf("Error in copying file: %v", err)
+		return err
+	}
+
+	cmd := exec.Command("tar", "-xvf", "/home/backup.tar", "--directory", "/home/")
+	var errMsg bytes.Buffer
+	cmd.Stderr = &errMsg
+
+	err = cmd.Run()
+	if err != nil {
+		log.Errorf(errMsg.String())
+		return err
+	}
+
+	// Change folder owner in order to have access to the folder locally
+	uID := viper.GetString("uid")
+	cmd = exec.Command("chown", "-R", uID, "/home")
+	cmd.Stderr = &errMsg
+
+	err = cmd.Run()
+	if err != nil {
+		log.Errorf(errMsg.String())
+		return err
+	}
+
+	err = d.Close()
+	if err != nil {
+		log.Errorf("Could not close decompressor: %v", err)
+	}
+
+	return nil
+}
+
```

Comment:
Base restore is non functional!
All the local tar stuff is just mocking and not related to the actual usecase.
There is no connector to the database server that should be restored

The backup command should need the backupfile to be restored and the address and credentials for the database

Reply:
I am not sure that I understand what are you saying here. Just need to mention that I have changed the names of the function and action to avoid confusion about restoration